app/controllers/courses_controller.rb

```diff
@@ -12,21 +18,43 @@ def show
   end
 
   def update
+    if !current_user.admin?
+      head :forbidden
+      return
+    end
+
+    # cannot update someone else's course
     course = Course.find(params[:id])
+    if course.teacher != current_user
```

Comment:
compare `course.teacher_id` with `current_user.id`. Rails here will make an additional query to the db to fetch the teacher object.